sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/LoggingUtils.java

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.logging;
+
+import com.azure.core.util.CoreUtils;
+
+/**
+ * Contains utility methods for logging.
+ */
+public class LoggingUtils {
+    private static final char CL = '\r';
+    private static final char RF = '\n';
+
+    /*
+     * Removes CRLF pattern in the {@code logMessage}.
+     *
+     * @param logMessage The log message to sanitize.
+     * @return The updated logMessage.
+     */
+    public static String sanitizeLogMessageInput(String logMessage) {
+        if (CoreUtils.isNullOrEmpty(logMessage)) {
+            return logMessage;
+        }
+
+        StringBuilder sb = null;
+        int prevStart = 0;
+
+        for (int i = 0; i < logMessage.length(); i++) {
+            if (logMessage.charAt(i) == CL || logMessage.charAt(i) == RF) {
+                if (sb == null) {
+                    sb = new StringBuilder(logMessage.length());
+                }
+
+                if (prevStart != i) {
+                    sb.append(logMessage, prevStart, i);
+                }
+                prevStart = i + 1;
+            }
```

Comment:
Given the amount of substringing going on I wonder if it would be more efficient for the first instance of a line feed (`\r\n` or `\n`) to trigger initializing the `StringBuilder` with the message until that point and then from there on out it copies character by character.